### PR TITLE
jsonnet/grafana: re-map datasources and config

### PR DESCRIPTION
**Pull request by paulfantom**
I need 2 more options to be mapped :(

Follow-up to https://github.com/prometheus-operator/kube-prometheus/pull/904

/cc @prometheus-operator/kube-prometheus-reviewers

### CODE REVIEW

File: jsonnet/kube-prometheus/components/grafana.libsonnet [hunk 1]
```diff
@@ -54,10 +56,17 @@ function(params) {
...
+      } + (
+        // conditionally overwrite default setting
+        if std.length(g.config.datasources) > 0 then
+        { datasources: g.config.datasources }
+        else {}
+      )
     },
   },
 
+  config: glib.grafana.config,
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/components/grafana.libsonnet:69]: I will take care of those mapping after I finish testing kube-prometheus inclusion with https://github.com/openshift/cluster-monitoring-operator/pull/1044

File: jsonnet/kube-prometheus/components/grafana.libsonnet [hunk 2]
```diff
@@ -26,45 +26,55 @@ local defaults = {
...
+  datasources: [],
+  config: {},
 };
 
 function(params) {
   local g = self,
-  config:: defaults + params,
-  //local g.config = defaults + params,
+  cfg:: defaults + params,
+  //local g.cfg = defaults + params,
```
Comment (lilic) [inline jsonnet/kube-prometheus/components/grafana.libsonnet:36]: Why do we have this comment?

File: jsonnet/kube-prometheus/components/grafana.libsonnet [hunk 3]
```diff
@@ -26,45 +26,55 @@ local defaults = {
...
-      },
+        labels: g.cfg.commonLabels,
+        dashboards: g.cfg.dashboards,
+        resources: g.cfg.resources,
+        rawDashboards: g.cfg.rawDashboards,
+        folderDashboards: g.cfg.folderDashboards,
+        containers: g.cfg.containers,
+        config+: g.cfg.config,
+      } + (
+        // conditionally overwrite default setting
```
Comment (lilic) [inline jsonnet/kube-prometheus/components/grafana.libsonnet:61]:
```suggestion
        // Conditionally overwrite default setting.
```

**Review by lilic — COMMENTED** [review]
One question, but lgtm

File: jsonnet/kube-prometheus/components/grafana.libsonnet (hunk 2 above)
Comment (paulfantom) [reply jsonnet/kube-prometheus/components/grafana.libsonnet]: Leftover from testing, I removed it.

File: jsonnet/kube-prometheus/components/grafana.libsonnet (hunk 3 above)
Comment (paulfantom) [reply jsonnet/kube-prometheus/components/grafana.libsonnet]: Thanks!

**Review by lilic — APPROVED** [review]
lgtm :shipit:

**Pull request merged**